### PR TITLE
[spark] refine MergeInto UT

### PR DESCRIPTION
**Pull request by YannByron**
### Purpose

Linked issue: close #xxx

### CODE REVIEW

File: paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/PaimonTableTest.scala [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+  def appendPrimaryKey(primaryKeys: Seq[String], props: mutable.Map[String, String]): Unit
+
+  def createTable(
+      tableName: String,
+      columns: String,
+      primaryKeys: Seq[String],
+      partitionKeys: Seq[String] = Seq.empty,
+      props: Map[String, String] = Map.empty): Unit = {
+    val newProps: mutable.Map[String, String] = mutable.Map.empty[String, String] ++ bucket ++ props
```
Comment (Zouxxyy) [inline paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/PaimonTableTest.scala:37]: What if props contains `bucket`? Should it be banned.

File: paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/PaimonTableTest.scala [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+package org.apache.paimon.spark
+
+import org.apache.spark.sql.test.SharedSparkSession
+
+import scala.collection.mutable
+
+trait PaimonTableTest extends SharedSparkSession {
+
+  val bucket: Map[String, String]
```
Comment (Zouxxyy) [inline paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/PaimonTableTest.scala:27]: can just use int

File: paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/PaimonTableTest.scala (hunk 1 above)
Comment (YannByron) [reply paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/PaimonTableTest.scala:37]: then should respect the props's bucket.

**Review by Zouxxyy — APPROVED** [review]
+1

**Pull request merged**